components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java

```diff
@@ -119,7 +119,8 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
                 JSONArray redirectUris = (JSONArray) obj;
                 for (Object redirectUri : redirectUris) {
                     try {
-                        if (IdentityUtil.isNotBlank((String) redirectUri) && !redirectUri.equals("null")) {
+                        if (redirectUri != null && IdentityUtil.isNotBlank((String) redirectUri) &&
+                                !"null".equals(redirectUri)) {
```

Comment:
casting a null value to String doesnt cause a class cast exception AFAIK. Therefore the first null check is not necessary. Also IdentityUtil.isNotBlank() covers  !"null".equals() check